client/js/widgets/pile.js

```diff
@@ -1,3 +1,18 @@
+// Compare the relative z order of two widgets with the same parent.
+function compareChildStackOrder(a, b) {
+  if (a == b)
+    return 0;
+  const aZ = a.z || 0;
+  const bZ = b.z || 0;
+  if (aZ != bZ)
+    return bZ - aZ;
+  let domOrder = a.domElement.compareDocumentPosition(b.domElement);
```

Comment:
I'm actually no longer sure if comparing dom order is important or if piles will in actual use always have unique z values. The test failure was unrelated.

Reply:
The only way widgets have the same computed z value is if they are set that way by setting z in an editor or using SET in a routine (and the layer properties are the same). This would be pretty rare in piles; I don't think accounting for DOM order is needed, but it doesn't hurt to handle this edge case.